src/Libraries/Nop.Services/ExportImport/ExportManager.cs

```diff
@@ -344,38 +344,20 @@ protected virtual string GetProductTags(Product product)
         }
 
         /// <summary>
-        /// Returns the three first image associated with the product
+        /// Returns the image at specified index associated with the product
         /// </summary>
         /// <param name="product">Product</param>
-        /// <returns>three first image</returns>
-        protected virtual string[] GetPictures(Product product)
+        /// <param name="pictureIndex">Picture index to get</param>
+        /// <returns>image thumb local path</returns>
+        protected virtual string GetPicture(Product product, short pictureIndex)
         {
-            //pictures (up to 3 pictures)
-            string picture1 = null;
-            string picture2 = null;
-            string picture3 = null;
-            var pictures = _pictureService.GetPicturesByProductId(product.Id, 3);
-            for (var i = 0; i < pictures.Count; i++)
-            {
-                var pictureLocalPath = _pictureService.GetThumbLocalPath(pictures[i]);
-                switch (i)
-                {
-                    case 0:
-                        picture1 = pictureLocalPath;
-                        break;
-                    case 1:
-                        picture2 = pictureLocalPath;
-                        break;
-                    case 2:
-                        picture3 = pictureLocalPath;
-                        break;
-                }
-            }
-
-            return new[] { picture1, picture2, picture3 };
+            // we need only the picture at a specific index, no need to get more pictures than that
+            var recordsToReturn = pictureIndex + 1;
+            var pictures = _pictureService.GetPicturesByProductId(product.Id, recordsToReturn);
+            return pictures.Count > pictureIndex ? _pictureService.GetThumbLocalPath(pictures[pictureIndex]) : null;
         }
 
-        protected virtual bool IgnoreExportPoductProperty(Func<ProductEditorSettings, bool> func)
+        protected virtual bool IgnoreExportProductProperty(Func<ProductEditorSettings, bool> func)

```

Comment:
Hi @roman-gherman. Thank you for the work done, but in the future, please do not mix two different refactorings, it is more difficult to analyze the code. In this case, a typo was not worth correcting.

Reply:
@skoshelev yes, makes sense